pkg/version/version_test.go

```diff
@@ -0,0 +1,155 @@
+		},
+		{
+			name:            "Unreleased version",
+			Version:         "1.10",
+			ReleaseStatus:   "unreleased",
+			expectedversion: "1.10-unknown",
+		},
+		{
+			name:            "Unreleased version with GitTreeState dirty",
+			Version:         "1.10",
+			ReleaseStatus:   "unreleased",
+			GitSHA:          "abc",
+			GitTreeState:    "dirty",
+			expectedversion: "1.10-abc.dirty",
+		},
+		{
+			name:            "Unreleased version with GitTreeState empty",
+			Version:         "1.10",
+			ReleaseStatus:   "unreleased",
+			GitSHA:          "abc",
+			GitTreeState:    "",
+			expectedversion: "1.10-abc",
+		},
+		{
+			name:            "Unreleased version with GitTreeState clean",
+			Version:         "1.10",
+			ReleaseStatus:   "unreleased",
+			GitSHA:          "abc",
+			GitTreeState:    "clean",
+			expectedversion: "1.10-abc",
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			Version = tc.Version
+			ReleaseStatus = tc.ReleaseStatus
+			GitSHA = tc.GitSHA
+			GitTreeState = tc.GitTreeState
+			gotVersion := GetFullVersion()
+			Version = ""
+			ReleaseStatus = ""
+			GitSHA = ""
+			GitTreeState = "unreleased"
+			assert.Equal(t, tc.expectedversion, gotVersion)
+		})
+	}
+}
+
+func TestGetFullVersionWithRuntimeInfo(t *testing.T) {
+	testCases := []struct {
+		name            string
+		Version         string
+		ReleaseStatus   string
+		GitSHA          string
+		GitTreeState    string
+		expectedversion string
+	}{
+		{
+			name:            "Unknown version",
+			expectedversion: fmt.Sprintf("UNKNOWN %s/%s %s", runtime.GOOS, runtime.GOARCH, runtime.Version()),
+		},
+		{
+			name:            "Released version",
+			Version:         "1.9",
+			ReleaseStatus:   "released",
+			expectedversion: fmt.Sprintf("1.9 %s/%s %s", runtime.GOOS, runtime.GOARCH, runtime.Version()),
+		},
+		{
+			name:            "Unreleased version",
+			Version:         "1.10",
+			ReleaseStatus:   "unreleased",
+			expectedversion: fmt.Sprintf("1.10-unknown %s/%s %s", runtime.GOOS, runtime.GOARCH, runtime.Version()),
+		},
+		{
+			name:            "Unreleased version with GitTreeState dirty",
+			Version:         "1.10",
+			ReleaseStatus:   "unreleased",
+			GitSHA:          "abc",
+			GitTreeState:    "dirty",
+			expectedversion: fmt.Sprintf("1.10-abc.dirty %s/%s %s", runtime.GOOS, runtime.GOARCH, runtime.Version()),
+		},
+		{
+			name:            "Unreleased version with GitTreeState empty",
+			Version:         "1.10",
+			ReleaseStatus:   "unreleased",
+			GitSHA:          "abc",
+			GitTreeState:    "",
+			expectedversion: fmt.Sprintf("1.10-abc %s/%s %s", runtime.GOOS, runtime.GOARCH, runtime.Version()),
+		},
+		{
+			name:            "Unreleased version with GitTreeState clean",
+			Version:         "1.10",
+			ReleaseStatus:   "unreleased",
+			GitSHA:          "abc",
+			GitTreeState:    "clean",
+			expectedversion: fmt.Sprintf("1.10-abc %s/%s %s", runtime.GOOS, runtime.GOARCH, runtime.Version()),
+		},
+	}
```

Comment:
These subtests are redundant with the ones in TestGetFullVersion. Since the focus is GetFullVersionWithRuntimeInfo instead of GetFullVersion and there is no other variable, I think you don't need the table here, the lines below can already provide same coverage:
```
got := GetFullVersionWithRuntimeInfo()
assert.Equal(t, fmt.Sprintf("...", ...), got)
```